04-spatial-operations.Rmd

```diff
@@ -41,53 +46,52 @@ It is important to note that spatial operations that use two spatial objects rel
 
 In general terms, spatial subsetting is simply the spatial equivalent of *attribute subsetting*.
-However, to do spatial subsetting *two spatial objects are needed* the spatial relation between which is to be established.
 As with attribute subsetting, spatial subsetting is a *binary operation*: an object is either selected or not.
-As in section \@ref(vector-attribute-subsetting), we start with base methods before describing how to do it in the **tidyverse**.
+Following the structure of section \@ref(vector-attribute-subsetting), we start with base methods before describing how to do it in the **tidyverse**.
 <!-- todo: link to non-binary links, e.g. area-weighted spatial interpolation -->
 
 <!-- ### Spatial subsetting in base R -->
 
-Attribute subsetting in base R is done with the `[` operator and passing into the square brackets a vector of class `integer` (whole numbers) or `logical` (a vector of `TRUE`s and `FALSE`s).
-This means `world[1:6, ]` subsets the first 6 countries of the world and that `world[world$area_km2 < 10000, ]` returns the subset of countries that have a small surface area.
-For this chapter we will use countries in Africa, which can be generated using this method as
-follows:^[Recall
+To introduce spatial vector subsetting operations we will use countries in Africa.
+We will apply attribute subsetting to the `world` dataset (see previous chapter):^[Recall
 that we can also subset simple features using the `filter()` function, e.g. with `filter(world, continent == "Africa")`]
 
 ```{r}
 africa_wgs = world[world$continent == "Africa", ]
 ```
 
-To further set-up the input data, we will reproject the data to the coordinate reference system (CRS) 32630 (it's EPSG code, explained in Chapter 6):
+To further set up the input data, we will reproject the data to the coordinate reference system (CRS) 32630, its EPSG code (explained in Chapter 6):
 
 ```{r}
 africa = st_transform(africa_wgs, crs = 32630)
 ```
 
-*Spatial* subsetting in base R use the same method as attribute subsetting, except *another spatial object* is placed inside the square brackets in the place of an `integer` or `logical` vector.
+We can also use the `[` operator for *Spatial* subsetting.
+The difference is that we use *another spatial object* inside the square brackets instead of an `integer` or `logical` vector.
 This is a concise and consistent syntax, as shown in the next code chunk.
 Let's test it with a hypothetical scenario: we want to subset all countries within 20 degrees of the point where the equator (where latitude = 0 degrees) intersects the prime meridian (longitude = 0 degrees), as illustrated in Figure \@ref(fig:globe).
 The subsetting object is created below.
 Note that this must have the same CRS as the target object (set with the `crs` argument):
 
 ```{r, warning=FALSE}
-center = st_sf(st_sfc(st_point(c(0, 0)), crs = 4326))
+center = st_sf(geometry = st_sfc(st_point(c(0, 0)), crs = 4326))
```

Comment:
Yes, important to be explicit. Thanks for that.

Reply:
the point here is, if you don't name the column `geometry`, it will be named automatically in a weird way.